ts/components/conversation/right-panel/overlay/disappearing-messages/TimeOptions.tsx

```diff
@@ -0,0 +1,44 @@
+import { isEmpty } from 'lodash';
+import React from 'react';
+import { TimerOptionsArray } from '../../../../../state/ducks/timerOptions';
+import { PanelButtonGroup, PanelLabel } from '../../../../buttons/PanelButton';
+import { PanelRadioButton } from '../../../../buttons/PanelRadioButton';
+
+type TimerOptionsProps = {
+  options: TimerOptionsArray | null;
+  selected?: number;
+  setSelected: (value: number) => void;
+  hasOnlyOneMode?: boolean;
+  disabled?: boolean;
+};
+
+export const TimeOptions = (props: TimerOptionsProps) => {
+  const { options, selected, setSelected, hasOnlyOneMode, disabled } = props;
+
+  if (!options || isEmpty(options)) {
+    return null;
+  }
+
+  return (
+    <>
+      {!hasOnlyOneMode && <PanelLabel>{window.i18n('timer')}</PanelLabel>}
+      <PanelButtonGroup>
+        {options.map((option: any) => {
```

Comment:
why the any? it's not typed right as a   `options: TimerOptionsArray | null;`?

Reply:
My mistake. It should be the `TimerOptionsEntry` type which I have now exported and imported.